### DISCUSSION TITLE
Is it possible to constrain model output gradients?

### DISCUSSION THREAD

**Discussion by vedadb** [meta-pytorch/botorch#3341] [Q&A]
Hi,

I have a system I want to model where the gradient dy/dxi (y=output, xi one of the input variables) should always be positive. Is it possible to add this model restriction in the model training somehow?

Best regards
Vedad

**Comment by esantorella** [comment]
You can use a linear kernel with the `Warp` input transform to enforce monotonicity (with the minor issue that technically you could get negative monotonicity instead of positive monotonicity if there's no clear trend in your data).
